src/engine/source/base/parseEvent.cpp

```diff
@@ -50,7 +51,14 @@ Event parseOssecEvent(const std::string& event)
                                              "to be right after the location"));
     }
 
-    parseEvent->setString(event.substr(LOCATION_OFFSET, locationIdx - LOCATION_OFFSET), EVENT_LOCATION_ID);
+    std::string unescapedLocation = event.substr(LOCATION_OFFSET, locationIdx - LOCATION_OFFSET);
+    for (auto j = unescapedLocation.find("|:", 0); j != std::string::npos; j = unescapedLocation.find("|:", j))
+    {
+        unescapedLocation.erase(j, 1);
+        j++;
+    }
+
+    parseEvent->setString(unescapedLocation, EVENT_LOCATION_ID);
```

Comment:
Maybe something like this would avoid duplicating the delimiter, what do you think?
```suggestion
    std::string location = event.substr(LOCATION_OFFSET, locationIdx - LOCATION_OFFSET);
    {
        size_t pos;
        while ((pos = location.find("|:")) != std::string::npos) {
            location.erase(pos, 1);
        }
    }
    parseEvent->setString(location, EVENT_LOCATION_ID);
```